### PR TITLE
Issue #447 Default behaviour change for batch error handling in 5.7.0-beta

### PR DESCRIPTION
**Pull request by VikingsFan**

### CODE REVIEW

**Comment by VikingsFan** [comment]
@chgarg we found last change for 'continue-on-error' is a breaking change (before 5.7 bata, this header is true by default, now is false), so we decide to add an flag in server side to decide turn on that or not, if you have time, would you please try it out and let us know any issues. Thanks.

File: OData/src/System.Web.OData/OData/Batch/DefaultODataBatchHandler.cs [hunk 1]
```diff
@@ -42,7 +44,12 @@ public DefaultODataBatchHandler(HttpServer httpServer)
             IList<ODataBatchRequestItem> subRequests = await ParseBatchRequestsAsync(request, cancellationToken);
 
             string preferHeader = RequestPreferenceHelpers.GetRequestPreferHeader(request);
-            if (preferHeader != null && preferHeader.Contains(PreferenceContinueOnError))
+            var configuration = request.GetConfiguration();
+            if (configuration == null)
+            {
+                throw Error.InvalidOperation(SRResources.RequestMustContainConfiguration);
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Batch/DefaultODataBatchHandler.cs]: Can we not throw this exception and let it do the following if configuration == null

File: OData/test/UnitTest/System.Web.OData.Test/OData/Batch/UnbufferedODataBatchHandlerTest.cs [hunk 2]
```diff
@@ -192,7 +206,7 @@ public void ProcessBatchAsync_ContinueOnError()
             });
             UnbufferedODataBatchHandler batchHandler = new UnbufferedODataBatchHandler(server);
             HttpRequestMessage batchRequest = new HttpRequestMessage(HttpMethod.Post, "http://example.com/$batch")
-            {
+            {
```
Comment (xuzhg) [inline OData/test/UnitTest/System.Web.OData.Test/OData/Batch/UnbufferedODataBatchHandlerTest.cs]: why add this?

**Comment by xuzhg** [comment]
Others look good to me。

**Comment by VikingsFan** [comment]
merged 9cbe805c0d75721d8ea5530fa7c12546c41aa3b9

**Pull request closed without merging**